README.md

```diff
@@ -44,15 +43,16 @@ card.number = "4111111111111111"
 card.cvv = "123"
 
 // STEP 4: Call checkout method
-cardClient.checkoutWithCard(ORDER_ID, card) { result, error _>
-    error?.let {
-        // handle checkout error
-        return
-    }
-    result?.let {
-        val orderID = it.orderID 
-        // Send orderID to your server to process the payment
-    }
+lifecycleScope.launch {
```

Comment:
Just for a sanity check - what does invoking a suspend function that throws an error look like in java?

Reply:
i think that for Java we might want to wrap the suspend function and use a callback. A merchant that uses Java would call `approveOrder` with a callback and we jsut wrap the suspend functions inside of it.
From what i read, callings Kotlin suspend functions from Java is not that trivial

Reply:
Should that be a part of this PR? Thinking about the next task of doing the Java demo app - will that be possible with the code as is?

Reply:
We do have a separate spike planned for providing callback extensions. We may end up with a Kotlin client that can be invoked from Java.